pkg/basetypes.go

```diff
@@ -1,17 +1,20 @@
 package ocr2keepers
 
 import (
-	"encoding/hex"
 	"encoding/json"
 	"fmt"
-	"math/big"
 	"strings"
-
-	"github.com/ethereum/go-ethereum/crypto"
 )
 
 type UpkeepIdentifier []byte
 
+type UpkeepType uint8
+
+const (
+	ConditionTrigger UpkeepType = iota
+	LogTrigger
```

Comment:
Just thinking about this; won't the `0` value assigned to `ConditionTrigger` mean that invalid/incomplete `UpkeepIdentifier` instances on upkeeps will be treated as `ConditionTrigger` types? 🤔 Like should we avoid relying on "default" values to determine validity? Not sure if that's something we need to consider/care about when testing error scenarios but just wanted to put it out there!

Reply:
Good point

Reply:
good point to think through. Upkeep identifiers are uint256 and as long as they are in the range of 0 to 2^256-1 they should be valid.

With new trigger types we defined an encoding where 4th to 14th bytes should be 0 for the upkeepID and 15th byte should contain the uint8 type. However we have old upkeepIDs defined in the system which might not follow the above pattern, there we want default to be conditionals.

Logic on contract here https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/dev/automation/2_1/KeeperRegistryBase2_1.sol#L683-L694 

I haven't verified if the offchain logic matches this, will create a ticket for it